Lib/typing.py

```diff
@@ -2970,7 +2971,12 @@ def __new__(cls, typename, bases, ns):
                 raise TypeError(
                     'can only inherit from a NamedTuple type and Generic')
         bases = tuple(tuple if base is _NamedTuple else base for base in bases)
-        types = ns.get('__annotations__', {})
+        if "__annotations__" in ns:
+            types = ns["__annotations__"]
+        elif "__annotate__" in ns:
+            types = ns["__annotate__"](inspect.VALUE)
+        else:
+            types = {}
```

Comment:
It feels unfortunate that this kind of metaprogramming becomes more complicated, but perhaps it's not a common enough use case for it to warrant its own helper function in `inspect` or `types`

Reply:
Yes, I'll probably want to add a helper function for this somewhere. However, I'd like to focus this PR on the compiler changes (plus whatever else is needed to get tests passing); a future PR will focus on providing better Python-level abstractions.

Reply:
Note that for full PEP 649 support this code is not sufficient, since it uses VALUE semantics to evaluate the entire annotations dict. That's something I plan to improve in a later PR.